docs: update the getting started docs page (#825)
asyncapi/website: pages/docs/getting-started/index.md

date: 2019-02-16T13:56:52+01:00
weight: 0
categories:
- introduction
aliases:
- '/v1/guide/'
- '/v1/guide/index.html'
---

> Welcome to the AysncAPI documentation section where you get to learn about solutions to automate and formalize the documentation or code generation of your event-driven (micro)services.

Reviewer: So this introduction needs to be an introduction to the **tutorial**, not the _spec_. 😸  Let's update this section to detail **what this tutorial is going to teach us** and **what we are going to build**.

⭐ For a tutorial **introduction**, you want to describe:
- what the user will build (list technologies and end goal)
- overview of main steps to build it
- what the user will learn 

Let's go ahead and give this section another go 😃

Likewise, If you are aiming to establish solid standards for your events and improve the governance of your asynchronous APIs, you are in the right place.

This tutorial will walk you through how to get started with AsyncAPI. 

## Background Context

AsyncAPI is an open source initiative that seeks to improve the current state of Event-Driven Architecture (EDA). Our long-term goal is to make working with EDAs as easy as working with REST APIs. That goes from documentation to code generation, from discovery to event management. Most of the processes you apply to your REST APIs nowadays would be applicable to your event-driven/asynchronous APIs too.

To make this happen, the first step has been to create a specification that allows developers, architects, and product managers to define the interfaces of an async API. In the same way, [OpenAPI (formerly known as Swagger)](https://github.com/OAI/OpenAPI-Specification) does for REST APIs.

> The AsyncAPI specification settles the base for a greater and better tooling ecosystem for EDA's.

## Installation Guide

> In order to successfully set up AsyncAPI, you'll need to download a few things:

Reviewer: ```suggestion
Before you proceed to the next stage, you'll need to download a few things:
```

servers:
  swamp:
    url: localhost
    protocol: ws
channels:
  user/login:
    subscribe:
      summary: A practice to login a user to a system.    
      message:
        $ref: '#/components/messages/UserLoginDetails'
components:
  messages:
    UserLoginDetails:
      payload:
        type: object
        properties:
          userName:
            type: string
            description: A logged in user name.
          password:
            type: number
            format: number
            description: The logged in user password.
          `}
</CodeBlock>

From the specifications above, we defined the following fields:

- `asyncapi`: A specific version of the document we are authoring.

- `Info`: Aspects of the documentation that are human readable, such as a title, API version, a description that supports markdown, and a license.

- `servers`: API's endpoints that can be used to consume or publish messages.

- `channels`: The connection points, in this example a 'user/login' channel that accepts the user's login information.

- `components`: A field to hold various schemas for our specification, here the user name and password.

## Summary

> In this tutorial, we were able to learn how to setup an AsyncAPI document from scratch.

Reviewer: 😸  For the **Summary** section in our tutorials, we are writing:
- a summary of what the user just built and what technologies were used (again)
- an explanation of what the user learned to do 

A big part of improving this tutorial means both re-writing and adding new content ❤️ based on our new IA goals.

## Next Steps

> Now that you've completed this tutorial, check out our [Streetlights](https://www.asyncapi.com/docs/tutorials/streetlights) tutorial to learn how to create a basic practical example of a system that can turn on and off streetlights with AsyncAPI.

Reviewer: ```suggestion
Now that you've completed this tutorial, check out our [Streetlights](https://www.asyncapi.com/docs/tutorials/streetlights) tutorial to learn how to create a sample IoT system that turns streetlights on and off with AsyncAPI.
```
